test/library/env.go

```diff
@@ -73,9 +74,17 @@ func (e *TestEnv) ProxyEnv() []string {
 	return []string{"http_proxy=" + e.Proxy, "https_proxy=" + e.Proxy, "no_proxy=127.0.0.1"}
 }
 
+// memoizedTestEnvsByTest maps *testing.T pointers to *TestEnv. It exists so that we don't do all the
+// environment parsing N times per test and so that any implicit assertions happen only once.
+var memoizedTestEnvsByTest sync.Map //nolint: gochecknoglobals
```

Comment:
`*testing.T` is a valid map key?

Reply:
Pointers are valid map keys, even though it's not super common. I also considered using `t.Name()` as the map key but this seemed marginally more precise.